subprojects/core-api/src/main/java/org/gradle/api/Project.java

```diff
@@ -427,6 +427,15 @@ public interface Project extends Comparable<Project>, ExtensionAware, PluginAwar
      */
     Project getProject();
 
+    /**
+     * <p>Returns an immutable view of this project, safe for use with isolated projects.</p>
+     *
+     * @return This project as an {@link IsolatedProject}. Never returns null.
+     * @since 8.7
+     */
+    @Incubating
+    IsolatedProject getIsolated();
```

Comment:
My vote would be something like `getIsolatedView` or `asIsolated`

Reply:
Considering how they will be expressed in our DSLs, `project.isolated.name` is a clear winner for me compared to either `project.isolatedView.name` or `project.asIsolated.name`

Reply:
And just to be perfectly transparent, this choice was the result of a long deliberation within the the team.